sdk/communication/Azure.Communication.Rooms/src/RoomsClient.cs

```diff
@@ -506,6 +461,32 @@ public virtual async Task<Response> RemoveParticipantsAsync(string roomId, IEnum
             }
         }
 
+#nullable enable
+        private static Dictionary<string, ParticipantProperties?> ConvertRoomParticipantsToDictionaryForUpsert(IEnumerable<RoomParticipant> participants)
+        {
+            var participantDictionary = new Dictionary<string, ParticipantProperties?>() { };
+#nullable disable
+            participants?.ToList().ForEach(participant =>
+            {
+                participantDictionary.Add(participant.CommunicationIdentifier.RawId, new ParticipantProperties() { Role = participant.Role });
```

Comment:
participant.CommunicationIdentifier.RawId - wouldn't this be participant.RawId ?